File: spectre-graphql-parser/src/test/java/de/netherspace/tools/spectre/utils/CustomMavenPluginLogger.java
package de.netherspace.tools.spectre.utils;

import lombok.extern.slf4j.Slf4j;
import org.apache.maven.plugin.logging.Log;

@Slf4j
public class CustomMavenPluginLogger implements Log {

    @Override
    public boolean isDebugEnabled() {
        return true;
    }

    @Override
    public void debug(CharSequence charSequence) {
        logger.debug(charSequence.toString());
    }

    @Override
    public void debug(CharSequence charSequence, Throwable throwable) {
        logger.debug(charSequence.toString(), throwable);
    }

    @Override
    public void debug(Throwable throwable) {
        logger.debug("", throwable);
    }

    @Override
    public boolean isInfoEnabled() {
        return true;
    }

    @Override
    public void info(CharSequence charSequence) {
        logger.info(charSequence.toString());
    }

    @Override
    public void info(CharSequence charSequence, Throwable throwable) {
        logger.info(charSequence.toString(), throwable);
    }

    @Override
    public void info(Throwable throwable) {
        logger.info("", throwable);
    }

    @Override
    public boolean isWarnEnabled() {
        return true;
    }

    @Override
    public void warn(CharSequence charSequence) {
        logger.warn(charSequence.toString());
    }

    @Override
    public void warn(CharSequence charSequence, Throwable throwable) {
        logger.warn(charSequence.toString(), throwable);
    }

    @Override
    public void warn(Throwable throwable) {
        logger.warn("", throwable);
    }

    @Override
    public boolean isErrorEnabled() {
        return true;
    }

    @Override
    public void error(CharSequence charSequence) {
        logger.error(charSequence.toString());
    }

    @Override
    public void error(CharSequence charSequence, Throwable throwable) {
        logger.error(charSequence.toString(), throwable);
    }

    @Override
    public void error(Throwable throwable) {
        logger.error("", throwable);
    }
}

File: spectre-graphql-parser/src/test/kotlin/de/netherspace/tools/spectre/utils/CustomMavenPluginLogger.kt
package de.netherspace.tools.spectre.utils

import org.apache.maven.plugin.logging.Log
import org.slf4j.LoggerFactory

class CustomMavenPluginLogger : Log {

    private val logger = LoggerFactory.getLogger(CustomMavenPluginLogger::class.java)

    override fun warn(p0: CharSequence?) {
        logger.warn(p0!!.toString())
    }

    override fun warn(p0: CharSequence?, p1: Throwable?) {
        logger.warn(p0!!.toString(), p1!!)
    }

    override fun warn(p0: Throwable?) {
        logger.warn("", p0!!)
    }

    override fun info(p0: CharSequence?) {
        logger.info(p0!!.toString())
    }

    override fun info(p0: CharSequence?, p1: Throwable?) {
        logger.info(p0!!.toString(), p1!!)
    }

    override fun info(p0: Throwable?) {
        logger.info("", p0!!)
    }

    override fun isInfoEnabled(): Boolean = true

    override fun isErrorEnabled(): Boolean = true

    override fun isWarnEnabled(): Boolean = true

    override fun error(p0: CharSequence?) {
        logger.error(p0!!.toString())
    }

    override fun error(p0: CharSequence?, p1: Throwable?) {
        logger.error(p0!!.toString(), p1!!)
    }

    override fun error(p0: Throwable?) {
        logger.error("", p0!!)
    }

    override fun isDebugEnabled(): Boolean = true

    override fun debug(p0: CharSequence?) {
        logger.debug(p0!!.toString())
    }

    override fun debug(p0: CharSequence?, p1: Throwable?) {
        logger.debug(p0!!.toString(), p1!!)
    }

    override fun debug(p0: Throwable?) {
        logger.debug("", p0!!)
    }
}
